### PR TITLE
feat: add run all nuts

### PR DESCRIPTION
**Pull request by peternhale**
add run all nuts

### CODE REVIEW

File: src/jobs/test-nut.yml [hunk 1]
```diff
@@ -20,10 +20,21 @@ parameters:
     description: 'By default, the latest version of the standalone CLI will be installed. To install via npm, supply a version tag such as "latest" or "6".'
     default: ''
     type: string
+  sfdx_executable_path:
```
Comment (peternhale) [inline src/jobs/test-nut.yml:23]: two additional parameters exposed, executable path which will be used to set `TESTKIT_EXECUTABLE_PATH` below.
repo_tag, used to checkout a specific version of the plugin. Should ensure that plugin test code is a close to the version of the plugin used by sfdx_cli.

File: src/jobs/test-nut.yml [hunk 2]
```diff
@@ -20,10 +20,21 @@ parameters:
...
+  repo_tag:
+    description: "The tag of the module repo to checkout, '' defaults to branch/PR"
+    default: ''
+    type: string
 
 executor:
   name: << parameters.os >>
 
+environment:
+  TESTKIT_EXECUTABLE_PATH: <<parameters.sfdx_executable_path>>
```
Comment (peternhale) [inline src/jobs/test-nut.yml:36]: Set testkit env var for executable path.

File: src/jobs/test-nut.yml [hunk 3]
```diff
@@ -32,6 +43,14 @@ steps:
       version: <<parameters.sfdx_version>>
       os: <<parameters.os>>
   - checkout
+  - when:
```
Comment (peternhale) [inline src/jobs/test-nut.yml:46]: Only run when repo_tag is set

File: src/jobs/test-nut.yml [hunk 4]
```diff
@@ -75,6 +94,8 @@ steps:
       name: Nuts
       command: |
         echo "Using node: $(node --version)"
+        echo "Environment Variables:"
```
Comment (peternhale) [inline src/jobs/test-nut.yml:97]: Dump env vars for posterity. The set env task done by circle only displays those env var made available via contexts and project settings.

**Review by RodEsp — APPROVED** [review]

File: src/jobs/release-lerna-packages.yml [hunk 5]
```diff
@@ -46,7 +46,9 @@ steps:
   - configure-github
   - restore_cache:
       keys:
+        - v3-npm-{{ CIRCLE_BRANCH }}-{{checksum "yarn.lock"}}
         - v3-npm-{{checksum "yarn.lock"}}
+        - v3-npm
```
Comment (peternhale) [inline src/jobs/release-lerna-packages.yml:51]: @amphro @RodEsp

Comment (amphro) [reply src/jobs/release-lerna-packages.yml:51]: 👍

Comment (amphro) [reply src/jobs/release-lerna-packages.yml:51]: Can you also change the `- save_cache:` on line 59 to include the branch?

**Review by amphro — APPROVED** [review]

File: src/jobs/release-lerna-packages.yml [hunk 6]
```diff
@@ -46,7 +46,9 @@ steps:
   - configure-github
   - restore_cache:
       keys:
+        - v3-npm-{{ CIRCLE_BRANCH }}-{{checksum "yarn.lock"}}
         - v3-npm-{{checksum "yarn.lock"}}
```
Comment (amphro) [inline src/jobs/release-lerna-packages.yml:50]: Actually, this is wrong. It needs to be 

```
        - v3-npm-{{ CIRCLE_BRANCH }}-{{checksum "yarn.lock"}}
        - v3-npm-{{ CIRCLE_BRANCH }}
        - v3-npm

```

**Review by amphro — CHANGES_REQUESTED** [review]

**Review by amphro — COMMENTED** [review]
Change to comment so you can merge without my review.

**Review by amphro — APPROVED** [review]

File: src/jobs/test-nut.yml [hunk 7]
```diff
@@ -67,15 +67,17 @@ steps:
       steps:
         - restore_cache:
             keys:
+              - v3-npm-{{ CIRCLE_BRANCH }}-{{checksum "yarn.lock"}}
               - v3-npm-{{checksum "yarn.lock"}}
```
Comment (amphro) [inline src/jobs/test-nut.yml:71]: I still think this line wants to be `- v3-npm-{{ CIRCLE_BRANCH }}`

Comment (peternhale) [reply src/jobs/test-nut.yml:71]: @amphro 
This was the keys you called out in slack
` We should change it to branch-yarnlock, then yarnlock, then just npm`

File: src/jobs/test-nut.yml [hunk 8]
```diff
@@ -33,26 +32,25 @@ parameters:
...
       version: <<parameters.sfdx_version>>
       os: <<parameters.os>>
-  - when:
-      condition:
-        equal: ['', <<parameters.repo_tag>>]
-        steps:
-          - checkout
+  - checkout
   - when:
       condition:
```
Comment (mshanemc) [inline src/jobs/test-nut.yml:47]: simpler `condition: <<parameters.repo_tag>>`

**Review by mshanemc — APPROVED** [review]

**Review by amphro — APPROVED** [review]

**Pull request merged**